services/state/roster/roster.proto

```diff
@@ -0,0 +1,125 @@
+syntax = "proto3";
+
+package com.hedera.hapi.node.state.roster;
+
+/*
+ * Hedera Network Services Protobuf
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import "basic_types.proto";
+
+option java_package = "com.hederahashgraph.api.proto.java";
+// <<<pbj.java_package = "com.hedera.hapi.node.state.roster">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * A single roster in the network state.
+ * <p>
+ * The roster SHALL be a list of `RosterEntry` objects.
+ */
+message Roster {
+
+    /**
+     * List of roster entries, one per consensus node.
+     * <p>
+     * This list SHALL contain roster entries in natural order of ascending node ids.
+     * This list SHALL NOT be empty.<br/>
+     */
+    repeated RosterEntry rosters = 1;
+}
+
+/**
+ * A single roster entry in the network state.
+ *
+ * Each roster entry in SHALL encapsulate the elements required
+ * to manage node participation in the Threshold Signature Scheme (TSS).<br/>
+ * All fields are REQUIRED.
```

Comment:
This contradicts the lines 92-93 just below. I suggest to remove this statement from here and instead use field-specific docs below about whether a field is required or not.

Reply:
how so, because it says it's optional on line 92? If yes, i don't think that's contradictory. Line 92/93 is just saying once it's set, you can't unset it.

Reply:
Well, it is. Line 50:
> All fields are REQUIRED.

Line 92:
> This field is OPTIONAL

So either the definition of `all` doesn't include this particular field for some reason, or not all fields are actually required. Logically, there's a contradiction currently. You either want to provide an explicit definition for the word `all` as used in this context, or you want to remove either of the statements.

As for "unsetting" a field - PBJ models are immutable by definition. So if you're talking about an instance of this model in JVM, then this can be assumed and doesn't need mentioning. If you're talking about the lifecycle of the RosterEntry object over a span of multiple roster changes, then I don't believe that this particular aspect belongs to the specification of this protobuf model because it's not a property of this model, but rather a property of an algorithm that implements and supports the lifecycle of the roster as a whole, which is a very high-level concept compared to this tiny low-level model.

Reply:
Gotcha. How about `All fields except tss_encryption_key are REQUIRED.. Is that better?

Reply:
That would sound perfect!